content-scripts/src/modules/initialize.js

```diff
@@ -0,0 +1,121 @@
+import { extractColorsAsRootVars } from "./utilities/colors";
+import removeElement from "./utilities/removeElement";
+import throttle from "./utilities/throttle";
+
+// Function to reveal Search Filters
+const revealSearchFilters = () => {
+  // Get grandparent of advanced search
+  const advancedSearch = document.querySelector(
+    `[data-testid="searchFiltersAdvancedSearch"]`
+  );
+
+  if (advancedSearch) {
+    const searchFilters =
+      advancedSearch.parentElement.parentElement.parentElement;
+    if (!searchFilters.classList.contains("searchFilters")) {
+      searchFilters.classList = searchFilters.classList + " searchFilters";
+    }
+    return;
+  }
+};
+
+// Function to set search bar width to length of placeholder
+const searchBarWidthReset = () => {
+  const searchBar = document.querySelector(
+    '[data-testid="SearchBox_Search_Input"]'
+  );
+
+  if (
+    searchBar &&
+    !window.location.pathname.includes("/search") &&
+    !window.location.pathname.includes("/explore")
+  ) {
+    const searchBarPlaceholderWidth =
+      searchBar.getAttribute("placeholder").length;
+
+    searchBar.style.width = `${searchBarPlaceholderWidth + 1}ch`;
+  }
+};
+
+// Function to add main stylesheet
+export const addStylesheets = () => {
+  const head = document.querySelector("head");
+  const mainStylesheet = document.createElement("link");
+  const typefullyStylesheet = document.createElement("link");
+  const mainStylesheetGitHub = document.createElement("link");
+  const typefullyStylesheetGitHub = document.createElement("link");
+
+  mainStylesheet.rel = "stylesheet";
+  mainStylesheet.type = "text/css";
+  mainStylesheet.href = chrome.runtime.getURL("css/main.css");
+  head.appendChild(mainStylesheet);
+
+  mainStylesheetGitHub.rel = "stylesheet";
+  mainStylesheetGitHub.type = "text/css";
+  mainStylesheetGitHub.href =
+    "https://cdn.jsdelivr.net/gh/typefully/minimal-twitter@5/css/main.css";
+  head.appendChild(mainStylesheetGitHub);
+
+  typefullyStylesheet.rel = "stylesheet";
+  typefullyStylesheet.type = "text/css";
+  typefullyStylesheet.href = chrome.runtime.getURL("css/typefully.css");
+  head.appendChild(typefullyStylesheet);
+
+  typefullyStylesheetGitHub.rel = "stylesheet";
+  typefullyStylesheetGitHub.type = "text/css";
+  typefullyStylesheetGitHub.href =
+    "https://cdn.jsdelivr.net/gh/typefully/minimal-twitter@5/css/typefully.css";
+  head.appendChild(typefullyStylesheetGitHub);
+};
+
+// Function to start MutationObserver
+export const observe = () => {
+  const observer = new MutationObserver((mutationsList) => {
+    if (mutationsList.length) {
+      searchBarWidthReset();
+      revealSearchFilters();
+      addTypefullyPlug();
+      saveCurrentReplyToLink();
+      addTypefullyReplyPlug();
+      checkUrlForFollow();
+      checkHomeTimeline();
+      addWriterModeButton();
+      addListsButton();
+      setTimeout(() => {
+        addGrowButton();
+      });
+      extractColorsAsRootVars();
```

Comment:
@thomaswang can you move it to the first place? I think it's better to extract colors before injecting UI

Reply:
I'm referring to `extractColorsAsRootVars`

Reply:
👍 add comment to make sure we keep it at the top